### ISSUE TITLE
refactor(assimilation): use gaussx.quadratic_form for the prior cost term

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/plumax#40]
## Problem / Request
`build_cost_x` computes the prior term as
```python
Binv_dx = gx.solve(background_op, delta_x_flat)
prior = 0.5 * jnp.dot(delta_x_flat, Binv_dx)
```
which is exactly `0.5 * gx.quadratic_form(background_op, delta_x_flat)` — verified identical semantics (single structurally-dispatched solve, `x @ dispatch_solve(A, x)`) at `gaussx/_distributions/_gaussian.py:21-38`.

## Motivation
One-line behavior-preserving cleanup; keeps the cost function expressed in the same gaussx vocabulary as the rest of `assimilation/`.

## Implementation Steps
- [ ] Replace at `src/plumax/assimilation/cost.py:101-102`

## Definition of Done
- [ ] Existing FD-gradient and cost-value tests pass unchanged
- [ ] `make test && make lint && make typecheck` green

## Testing
- [ ] Existing `tests/assimilation/` suites suffice

## Relationships
- Parent (theme epic): #21
